### PR TITLE
Migrate codebase to PHP 8.1

### PR DESCRIPTION
**Pull request by derrabus**
Let's give this little library its PHP 8.1 treatment. ~~If this PR gets accepted, I'll create a 2.0.x branch and merge it there.~~

Replaces #44

### CODE REVIEW

File: src/Doctrine/Instantiator/Instantiator.php [hunk 1]
```diff
@@ -20,44 +22,40 @@
...
 {
     /**
      * Markers used internally by PHP to define whether {@see \unserialize} should invoke
      * the method {@see \Serializable::unserialize()} when dealing with classes implementing
      * the {@see \Serializable} interface.
      */
-    public const SERIALIZATION_FORMAT_USE_UNSERIALIZER   = 'C';
-    public const SERIALIZATION_FORMAT_AVOID_UNSERIALIZER = 'O';
+    private const SERIALIZATION_FORMAT_USE_UNSERIALIZER   = 'C';
+    private const SERIALIZATION_FORMAT_AVOID_UNSERIALIZER = 'O';
```
Comment (derrabus) [inline src/Doctrine/Instantiator/Instantiator.php:35]: Those constants are never used publicly and I believe they should not be part of the public interface. Do we need to flag them as deprecated on 1.x or is it enough to just switch them to private in 2.0?

Comment (greg0ire) [reply src/Doctrine/Instantiator/Instantiator.php:35]: Let's flag them as deprecated out of consistency with https://github.com/doctrine/orm/pull/10342?

**Review by greg0ire — APPROVED** [review]

File: src/Doctrine/Instantiator/Instantiator.php (hunk 1 above)
Comment (derrabus) [reply src/Doctrine/Instantiator/Instantiator.php:35]: Done in #96

File: src/Doctrine/Instantiator/Instantiator.php [hunk 2]
```diff
@@ -31,37 +31,33 @@ final class Instantiator implements InstantiatorInterface
      *
      * @deprecated This constant will be private in 2.0
```
Comment (stof) [inline src/Doctrine/Instantiator/Instantiator.php:32]: you forgot to remove that `@deprecated` tag

Comment (derrabus) [reply src/Doctrine/Instantiator/Instantiator.php:32]: Right. Wanna submit a PR?

**Pull request merged**